src/main/scala/uk/gov/homeoffice/drt/api/ArrivalRoutes.scala

```diff
@@ -0,0 +1,56 @@
+package uk.gov.homeoffice.drt.api
+
+import java.util.Date
+
+import cats.effect.Sync
+import cats.implicits._
+import org.http4s.dsl.Http4sDsl
+import org.http4s.util.CaseInsensitiveString
+import org.http4s.{HttpRoutes, Response, Status}
+import org.slf4j.LoggerFactory
+import uk.gov.homeoffice.drt.ResourceObject.getClass
+import uk.gov.homeoffice.drt.applicative.ArrivalFlights
+import uk.gov.homeoffice.drt.model.RequestedFlightDetails
+import uk.gov.homeoffice.drt.utils.DateUtil
+import uk.gov.homeoffice.drt.coders.ArrivalCoder._
+
+import scala.util.{Failure, Success, Try}
+
+object ArrivalRoutes {
+
+  private final val logger = LoggerFactory.getLogger(getClass.getName);
+
+  def arrivalFlightsRoutes[F[_] : Sync](H: ArrivalFlights[F], permissions: List[String]): HttpRoutes[F] = {
+    val dsl = new Http4sDsl[F] {}
+    import dsl._
+    HttpRoutes.of[F] {
+      case req@GET -> Root / "flights" / region :? params =>
+        Try {
+          val xAuthRoles: List[String] = req.headers.get(CaseInsensitiveString("X-Auth-Roles")).map(_.value.split(",").toList).getOrElse(List.empty)
+          val isAppropriatePermissionExists: Boolean = xAuthRoles.exists(p => permissions.contains(p))
+          isAppropriatePermissionExists match {
```

Comment:
this might read clearer as an `if (isAppropriatePermissionExists) {} else {}`

Reply:
If you prefer that I can change it .  In general in habit of avoiding if

Reply:
I'm fine with it if you prefer it that way.

I'd rename the val though as it doesn't make sense as it is. Maybe `userHasPermission`?

Also, `permissions` could be called something like `requiredPermissions` if that's what they are to make it clearer